### DISCUSSION TITLE
Datasets and models

### DISCUSSION THREAD

**Discussion by saed13** [saed13/AI-PROJECT-SMART_RECORDING_PIPELINE#2] [General]
ideas and analysis of the potential datasets and models

**Comment by Micropot** [comment]
- a2d2 datatset has 41k images already segmented with 38 different categories. We can use it for the recognition of the different elements of the street such as trafic lights or signs. Since this dataset as been created with real images, it's closer to the reality than a simulator, but it might not have any weird situations. 
- KITTI Dataset is a dataset similar to the a2d2 dataset but well known for the training of AI on the autonomous driving subject. On their website, they have lost of benchmark on lot of different type of data collected my a car with cameras on it.

**Comment by jzzzza** [comment]
a2d2 (Audi Autonomous Driving Dataset)
        Task: Autonomous driving
        Pros:
            High-resolution camera images and point clouds
            Accurate sensor data and ground truth annotations
            Data captured in diverse weather conditions and scenarios
        Cons:
            Limited to Audi vehicles and their specific sensors

    nuScenes
        Task: Autonomous driving
        Pros:
            Multi-modal dataset with camera, LiDAR, and RADAR data
            Large dataset with various urban driving scenarios
            Detailed annotations and object tracking
        Cons:
            Complex dataset structure
            Requires a significant amount of storage space and computational resources

    BDD100k (Berkeley DeepDrive)
        Task: Autonomous driving, scene understanding, and object detection
        Pros:
            Large-scale dataset with diverse driving scenarios
            Contains images, videos, and GPS/IMU data
            Annotations for object detection, segmentation, and lane marking
        Cons:
            No LiDAR or RADAR data
            Some annotations might be noisy or incomplete

    Waymo Open Dataset
        Task: Autonomous driving, object detection, and tracking
        Pros:
            High-quality and diverse dataset
            Multi-modal data with camera, LiDAR, and GPS/IMU
            Pre-processed sensor data for easier usage
        Cons:
            Large size requiring significant storage space
            Limited to Waymo vehicles and their specific sensors

    ApolloScape
        Task: Autonomous driving, semantic segmentation, and instance segmentation
        Pros:
            Large-scale dataset with diverse driving scenarios
            High-resolution images and videos
            Detailed annotations for object detection and segmentation
        Cons:
            No LiDAR or RADAR data
            Complex dataset structure

------------------------------------

    YOLO (You Only Look Once)
        Task: Object detection
        Pros:
            Real-time object detection with high accuracy
            Single-pass architecture for faster processing
        Cons:
            Struggles with small object detection

    Faster R-CNN
        Task: Object detection
        Pros:
            High accuracy in object detection
            Good for detecting small objects
        Cons:
            Slower than YOLO, not suitable for real-time applications

    EfficientDet
        Task: Object detection
        Pros:
            High accuracy and efficient architecture
            Scalable with different model sizes
        Cons:
            Slower than YOLO but faster than Faster R-CNN

    SegNet
        Task: Semantic segmentation
        Pros:
            Good for scene understanding tasks
            Efficient architecture with fast inference
        Cons:
            Less accurate than some other semantic segmentation models

    DeepLab
        Task: Semantic segmentation
        Pros:
            High accuracy in semantic segmentation tasks
            Incorporates multi-scale features for better performance
        Cons:
            Slower inference than some other semantic segmentation models

    Mask R-CNN
        Task: Instance segmentation
        Pros:
            High accuracy in object detection and segmentation
            Flexible architecture for various tasks
        Cons:
            Slower than other object detection models like YOLO

-----------------------------------------------------

    Object Detection: YOLOv5 or EfficientDet
    Semantic Segmentation: DeepLabv3+ or SegNet
    Object Tracking: DeepSORT or SORT

    BDD100k: Good for object detection, lane markings, and traffic infrastructure, but it contains data from the United States.
    ApolloScape: High-resolution images with semantic segmentation annotations, but data is from China.
    a2d2 : High-resolution data from Germany, suitable for detecting traffic infrastructure elements, but has limited samples.

**Comment by saed13** [comment]
https://github.com/FabianGermany/AutonomousDrivingDetectron2

**Comment by saed13** [comment]
https://github.com/facebookresearch/detectron2
